referenceframe/frame_system.go

```diff
@@ -308,6 +309,48 @@ func (sfs *simpleFrameSystem) FrameSystemSubset(newRoot Frame) (FrameSystem, err
 	return newFS, nil
 }
 
+// FrameSystemToPCD in a framesystem and returns a map where all elements are the point representation
+// of their geometry type with respect to the world.
+func FrameSystemToPCD(system FrameSystem, inputs map[string][]Input) (map[string][]r3.Vector, error) {
+	vectorMap := make(map[string][]r3.Vector)
+	for _, name := range system.FrameNames() {
+		currentFrame := system.Frame(name)
+		parent, err := system.Parent(currentFrame)
+		if err != nil {
+			return nil, err
+		}
+		currentInput := inputs[name]
+		if currentInput == nil {
+			currentInput = make([]Input, len(currentFrame.DoF()))
```

Comment:
Why do we want this?

Reply:
The question here is "If a frame has non-0 dof, do we display it in its zero position, or not display it at all?"

This would display it in its zero position.

I would be more inclined to not display it at all though.

Reply:
I think it would be a bad idea to assume it is in the zero position, so we should omit

Reply:
ah, I understand now. thanks for explaining

Reply:
If the item without any inputs has any children, this whole function will error on line 333 anyway.

Reply:
Following up on this I believe we want this
```		
currentInput := inputs[name]
if currentInput == nil && len(currentFrame.DoF()) == 0 {
	currentInput = []Input{}
}
```